src/main/java/com/github/mc1arke/sonarqube/plugin/CommunityBranchAgent.java

```diff
@@ -88,4 +91,37 @@ private static void redefineEdition(Instrumentation instrumentation) throws Clas
         instrumentation.retransformClasses(Class.forName(targetClassName));
     }
 
+    private static class MethodDetails {
+
+        private final String className;
+        private final String methodName;
+        private final String methodBody;
+
+        public MethodDetails(String className, String methodName, String methodBody) {
+            this.className = className;
+            this.methodName = methodName;
+            this.methodBody = methodBody;
+        }
+
+        public String getClassName() {
+            return className;
+        }
+
+        public String getMethodName() {
+            return methodName;
+        }
+
+        public String getMethodBody() {
+            return methodBody;
+        }
+    }
+
+    private enum Edition {
```

Comment:
I see this is a private inner class but, any reason to not make it static?

Reply:
Enums are always implicitly static so there's no need to explicitly declare them as static.